lib/cide/cli.rb

```diff
@@ -54,117 +56,158 @@ class CLI < Thor
       aliases: ['-s'],
       default: '~/.ssh/id_rsa'
 
-    def build
-      containers = []
-
+    def exec
       setup_docker
 
-      ## Config ##
+      tag = name_to_tag options.name
+
       banner 'Config'
-      build = Build::Config.load(Dir.pwd)
-      exit 1 if build.nil?
-      ssh_key = File.expand_path(options.ssh_key)
-      build.run = ['sh', '-e', '-c', options.run] unless options.run.empty?
-      name = CIDE::Docker.id options.name
-      tag = "cide/#{name}"
-      say_status :config, build.inspect
+      config = ConfigFile.load(Dir.pwd)
+      say_status :config, config.inspect
 
       ## Build ##
       banner 'Build'
-      if build.use_ssh
-        unless File.exist?(ssh_key)
-          fail ArgumentError, "SSH key #{ssh_key} not found"
-        end
-        create_tmp_file TEMP_SSH_KEY, File.read(ssh_key)
-      end
-      create_tmp_file DOCKERFILE, build.to_dockerfile
-      build_options = ['--force-rm']
-      build_options << '--pull' if options.pull
-      build_options.push '-f', DOCKERFILE
-      build_options.push '-t', tag
-      build_options << '.'
-      docker :build, *build_options
-
-      ## CI ##
+      builder = Builder.new(config)
+      builder.build(
+        pull: options.pull,
+        ssh_key: File.expand_path(options.ssh_key),
+        tag: tag,
+      )
+
+      ## Run ##
       banner 'Run'
-      build.links.each do |link|
-        args = ['--detach']
-        link.env.each_pair do |key, value|
-          args.push('--env', [key, value].join('='))
-        end
-        args << link.image
-        args << link.run if link.run
-        link.id = docker(:run, *args, capture: true).strip
-        containers << link.id
-      end
 
-      run_options = ['--detach']
+      command = options.run ? ['sh', '-e', '-c', options.run] : config.run
 
-      build.env.each_pair do |key, value|
-        run_options.push '--env', [key, value].join('=')
-      end
+      runner = Runner.new(
+        command: command,
+        env: config.env,
+        links: config.links,
+        tag: tag,
+      )
+      runner.run!
 
-      build.links.each do |link|
-        run_options.push '--link', [link.id, link.name].join(':')
-      end
+      ## Export ##
+      return unless options.export
+      banner 'Export'
+      runner.export!(
+        guest_dir: options.guest_export_dir || config.export_dir,
```

Comment:
If the user passes an export_dir it's to change the destination. If it also changes the source it would be confusing I think.